File: queue/python/queue.py
class Queue:
	def __init__(self):
		self.queue = []

	# Return true if the queue is empty false otherwise
	def isEmpty(self):
		return self.queue == []

	# Add item to the beginning of the queue
	def enqueue(self, item):
		self.queue.insert(0,item)
		print(str(item) + " has been enqueued.")

	# Remove and return the first item in the queue if not empty
	def dequeue(self):
		try:
			print(str(self.queue[-1]) + " has been dequeued.")
			return self.queue.pop()
		except IndexError:
			print("Can't dequeue from an empty queue.")

	# Return the front element of the queue if it exists
	def front(self):
		try:
			print(str(self.queue[-1]) + " is in the front.")
			return self.queue[-1]
		except:
			print("Queue is empty.  Nothing is in the front.")

	# Return the number of items in the queue
	def length(self):
		return len(self.queue)

	# Print the contents of the queue
	def print(self):
		print(self.queue)

	# Dequeue all items from the queue
	def clear(self):
		while not self.isEmpty():
			self.dequeue()
		print("The queue is clear.")


def main():
	my_queue = Queue()
	my_queue.front()
	my_queue.enqueue(5)
	my_queue.dequeue()
	my_queue.dequeue()
	my_queue.enqueue(3)
	my_queue.enqueue(8)
	my_queue.print()
	my_queue.front()
	my_queue.enqueue(2)
	my_queue.enqueue(4)
	my_queue.print()
	my_queue.clear()

if __name__ == "__main__":
	main()

File: queue/javascript/queue.js
class Queue {
    // Implememted using Array
    constructor() {
        this.items = [];
    }

    // Related Functions

    // enqueue function 
    enqueue(element) {
        // adding element to the queue 
        this.items.push(element);
    }

    // dequeue function 
    dequeue() {
        // removing element from the queue 
        // returns underflow when called 
        // on empty queue 
        if (this.isEmpty())
            return "Underflow";
        return this.items.shift();
    }

    // front function 
    front() {
        // returns the Front element of 
        // the queue without removing it. 
        if (this.isEmpty())
            return "No elements in Queue";
        return this.items[0];
    }

    // isEmpty function 
    isEmpty() {
        // return true if the queue is empty. 
        return this.items.length == 0;
    }

    // printQueue function 
    printQueue() {
        var str = "";
        for (var i = 0; i < this.items.length; i++)
            str += this.items[i] + " ";
        return str;
    }
 
}

// Implementation

// creating object for queue class 
var queue = new Queue();


// Testing dequeue and pop on an empty queue 
// returns Underflow 
console.log(queue.dequeue());

// returns true 
console.log(queue.isEmpty());

// Adding elements to the queue 
queue.enqueue(10);
queue.enqueue(20);
queue.enqueue(30);
queue.enqueue(40);
queue.enqueue(50);
queue.enqueue(60);

// returns 10 
console.log(queue.front());

// removes 10 from the queue
console.log(queue.dequeue());

// returns 20 
console.log(queue.front());

// removes 20 
console.log(queue.dequeue());

// printing the elements of the queue
console.log(queue.printQueue());
